### PR TITLE
Feat: Convert admin/collections to typescript 

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/collections/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export const defaultAdminCollectionsFields = ["id", "title", "handle"]
+export const defaultAdminCollectionsRelations = ["products"]
+export type AdminListCollectionsResponse = {
+  collections: ProductCollection[]
+  count: number
+  offset: number
+  limit: number
+}
+export type AdminCollectionResponse = {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/collections/index.ts:29]:
```suggestion
export type AdminCollectionsRes = {
```

File: packages/medusa/src/api/routes/admin/collections/list-collections.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  const [collections, count] = await productCollectionService.listAndCount(
+    {},
+    listConfig
+  )
+
+  res.status(200).json({ collections, count, offset, limit })
+}
+
+export class AdminGetCollectionsReq {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/collections/list-collections.ts:54]:
```suggestion
export class AdminGetCollectionsParams {
```

File: packages/medusa/src/api/routes/admin/collections/index.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  route.get("/:id", middlewares.wrap(require("./get-collection").default))
+  route.get("/", middlewares.wrap(require("./list-collections").default))
+
+  return app
+}
+
+export const defaultAdminCollectionsFields = ["id", "title", "handle"]
+export const defaultAdminCollectionsRelations = ["products"]
+export type AdminListCollectionsResponse = {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/collections/index.ts:23]:
```suggestion
export type AdminCollectionsListRes = {
```

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/collections/index.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  route.get("/:id", middlewares.wrap(require("./get-collection").default))
+  route.get("/", middlewares.wrap(require("./list-collections").default))
+
+  return app
+}
+
+export const defaultAdminCollectionsFields = ["id", "title", "handle"]
+export const defaultAdminCollectionsRelations = ["products"]
+
+export type AdminCollectionsListRes = {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/collections/index.ts:24]: Use `PaginatedResponse` :)

File: packages/medusa/src/api/routes/admin/collections/list-collections.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+export class AdminGetCollectionsParams {
+  @IsNumber()
+  @IsOptional()
+  limit?: number
+
+  @IsNumber()
+  @IsOptional()
+  offset?: number
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/collections/list-collections.ts:61]: These need `@Type(() => Number)` decorator
```suggestion
  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  limit?: number

  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  offset?: number
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM

**Pull request merged**